cloud/wfm/src/task/BaseTask.ts

```diff
@@ -0,0 +1,64 @@
+import { EventEmitter } from 'eventemitter3';
+import {Result} from '../result';
+import { Task, TaskEventData, TaskStatus } from './index';
+
+/**
+ * Base implementation for {@link Task}s
+ * Derived classes are expected to implement their run {@link #run} methods
+ */
+class BaseTask extends EventEmitter implements Task {
+  public result: Result;
+  /**
+   * Storage field for the `status` property
+   */
+  protected _status: TaskStatus | number = TaskStatus.PENDING;
```

Comment:
property name is `status` as per the interface, this is the storage variable for it, we might need to decide on a naming scheme.

Compiled js relies on [`Object.defineProperty`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty), from the compatibility notes I think it's a feature we can use.